### PR TITLE
Support SHOW EXTENDED TABLES

### PR DESCRIPTION
**Pull request by OscarTHZhang**
I saw the other ones are pretty similar. But not sure if this one is the right way to implement

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6159)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (JLDLaughlin) [inline .vscode/settings.json:1]: I believe this was mistakenly included! Could you remove this change from your commit?

File: doc/user/content/sql/show-tables.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ instances.
 Field | Use
 ------|-----
 _schema&lowbar;name_ | The schema to show tables from. Defaults to `public` in the current database. For available schemas, see [`SHOW SCHEMAS`](../show-schemas).
+`EXTENDED` | Returns system types as well as user-created tables. By default, only user-created tables are returned.
```
Comment (JLDLaughlin) [inline doc/user/content/sql/show-tables.md:19]: Replace `Returns system types` with `Returns system tables`

File: src/sql/src/plan/statement/show.rs [hunk 3]
```diff
@@ -240,29 +240,28 @@ fn show_tables<'a>(
...
-
     let schema = if let Some(from) = from {
         scx.resolve_schema(from)?
     } else {
         scx.resolve_default_schema()?
     };
 
-    let query = if full {
-        format!(
+    let mut query = format!(
```
Comment (JLDLaughlin) [inline src/sql/src/plan/statement/show.rs:249]: In order for us to grab system tables, we'll need to join with `mz_schemas` like we do in `show_types`:

```
        SELECT t.name, mz_internal.mz_classify_object_id(t.id) AS type
        FROM mz_catalog.mz_types t
        JOIN mz_catalog.mz_schemas s ON t.schema_id = s.id
        WHERE t.schema_id = {}",
        schema.id(),
```
The `s` in your `extended` case is referring to `mz_catalog.mz_schemas`, but it's not accessible yet! To update that, and to match `show_types`, I think you can replace lines 249 - 263 with:
```
    let mut query = format!(
        "SELECT t.name, mz_internal.mz_classify_object_id(t.id) AS type
        FROM mz_catalog.mz_tables t
        JOIN mz_catalog.mz_schemas s ON t.schema_id = s.id
        WHERE schema_id = {}",
        schema.id(),
    );
    if extended {
        query += " OR s.database_id IS NULL";
    }
    if !full {
        query = format!("SELECT name FROM ({})", query);
    }
```

**Review by JLDLaughlin — COMMENTED** [review]
This looks great so far, Oscar! Just have a few suggestions and a request that you add the following test cases starting at [line 382 of `catalog.td`](https://github.com/MaterializeInc/materialize/blob/main/test/testdrive/catalog.td#L382):

```
> CREATE SCHEMA tester

> SHOW TABLES FROM tester

> SHOW EXTENDED tables FROM tester
mz_array_types
mz_avro_ocf_sinks
mz_base_types
mz_columns
mz_databases
mz_functions
mz_index_columns
mz_indexes
mz_kafka_sinks
mz_list_types
mz_map_types
mz_pseudo_types
mz_roles
mz_schemas
mz_sinks
mz_sources
mz_tables
mz_types
mz_view_foreign_keys
mz_view_keys
mz_views

> CREATE TABLE tester.test_table (a int)

> SHOW TABLES FROM tester
test_table

> SHOW EXTENDED tables FROM tester
mz_array_types
mz_avro_ocf_sinks
mz_base_types
mz_columns
mz_databases
mz_functions
mz_index_columns
mz_indexes
mz_kafka_sinks
mz_list_types
mz_map_types
mz_pseudo_types
mz_roles
mz_schemas
mz_sinks
mz_sources
mz_tables
mz_types
mz_view_foreign_keys
mz_view_keys
mz_views
test_table
```

File: src/sql/src/plan/statement/show.rs (hunk 3 above)
Comment (OscarTHZhang) [reply src/sql/src/plan/statement/show.rs:249]: Hi @JLDLaughlin, thanks for reviewing my PR. I've added my changes below. How do I re-run the build checks?

**Comment by JLDLaughlin** [comment]
hey @OscarTHZhang -- I missed one thing in my review! Could you remove the space after ["... `returned.` "](https://github.com/MaterializeInc/materialize/pull/6159/files#diff-c5f30049138f25166ecc30c5000e2cb6b392a152ba34824b0f334ba2f8c6d23eR19) in the docs? Once you do, I can retrigger the tests and they will all pass.

**Comment by OscarTHZhang** [comment]
> hey @OscarTHZhang -- I missed one thing in my review! Could you remove the space after ["... `returned.` "](https://github.com/MaterializeInc/materialize/pull/6159/files#diff-c5f30049138f25166ecc30c5000e2cb6b392a152ba34824b0f334ba2f8c6d23eR19) in the docs? Once you do, I can retrigger the tests and they will all pass.

Just did it!

**Comment by JLDLaughlin** [comment]
🎉 thanks @OscarTHZhang ! Just merged this for you

**Pull request merged**